applications/examples/example_number_input/example_number_input.c

```diff
@@ -0,0 +1,79 @@
+#include "example_number_input.h"
+
+bool example_number_input_custom_event_callback(void* context, uint32_t event) {
+    furi_assert(context);
+    ExampleNumberInput* app = context;
+    return scene_manager_handle_custom_event(app->scene_manager, event);
+}
+
+static bool example_number_input_back_event_callback(void* context) {
+    furi_assert(context);
+    ExampleNumberInput* app = context;
+    return scene_manager_handle_back_event(app->scene_manager);
+}
+
+static ExampleNumberInput* example_number_input_alloc() {
+    ExampleNumberInput* app = malloc(sizeof(ExampleNumberInput));
+    app->gui = furi_record_open(RECORD_GUI);
+
+    app->view_dispatcher = view_dispatcher_alloc();
+
+    app->scene_manager = scene_manager_alloc(&example_number_input_scene_handlers, app);
+    view_dispatcher_set_event_callback_context(app->view_dispatcher, app);
+    view_dispatcher_set_custom_event_callback(
+        app->view_dispatcher, example_number_input_custom_event_callback);
+    view_dispatcher_set_navigation_event_callback(
+        app->view_dispatcher, example_number_input_back_event_callback);
+
+    app->number_input = number_input_alloc();
+    view_dispatcher_add_view(
+        app->view_dispatcher,
+        ExampleNumberInputViewIdNumberInput,
+        number_input_get_view(app->number_input));
+
+    app->dialog_ex = dialog_ex_alloc();
+    view_dispatcher_add_view(
+        app->view_dispatcher,
+        ExampleNumberInputViewIdShowNumber,
+        dialog_ex_get_view(app->dialog_ex));
+
+    app->current_number = 5;
+    app->min_value = INT32_MIN;
```

Comment:
I've made couple changes in example code:
- Simplified it as much as possible
- Changed min max values to int32_t specification

Looks like you incorrectly handling INT32_MIN and INT32_MAX in number_input module. Try to input something bigger or smaller that INT32_MIN or INT32_MAX.

Reply:
There's a reason I used +/- 1 on the INT32_MIN/INT32_MAX specifications. 

You removed these lines and replaced it with the MIN/MAX. That's why it no longer works. 
`
int32_t max = INT32_MAX - 1; // 2147483646, limit has no effect if larger
int32_t min = INT32_MIN + 1; // -2147483647 limit has no effect if smaller
`

The module needs to make an int of the input string and compare it with the min/max values. If using the max allowed int32 values the number resets if bigger/smaller and the comparison doesn't work. 

If the solution of reducing the limit by 1 is not ok, please advise me how I can fix this in a better way.

Reply:
Great, thank you for the update and solution.